<launch>
    <arg name="record_bag" default="false" />
    <arg name="id" default="1" />
    <arg name="name" default="uav$(arg id)" />
    <arg name="phase" default="0.0" />
    <arg name="altitude" default="7.0" />
    <arg name="xc" default="0.0" />
    <arg name="yc" default="0.0" />

    <group ns="$(arg name)">
    <node pkg="px4_base" type="circle" name="circle_node" output="screen">
            <param name="altitude" value="$(arg altitude)"/>
            <param name="takeoff_time" value="10.0"/>
            <param name="rotation_frequency" value="0.1"/>
            <param name="radius" value="2.0"/>
            <param name="phase" value="$(arg phase)"/>
            <param name="id" value="$(arg id)" />
            <param name="center_x" value="$(arg xc)" />
            <param name="center_y" value="$(arg yc)" />
            <remap from="mavros/local_position/odom" to="/supervisor/uav$(arg id)/odom"/>
    </node>
    </group>

    <node pkg="rosbag" type="record" name="rosbag_record"
        if="$(arg record_bag)"
        args="record -a -o $(find px4_base)/bags/" />

</launch>
